src/dune_engine/process.ml

```diff
@@ -601,36 +603,51 @@ let run_internal ?dir ?(stdout_to = Io.stdout) ?(stderr_to = Io.stderr)
       | None, None -> ()
       | _, _ -> assert false);
       Option.iter response_file ~f:Path.unlink;
+      let actual_stdout =
+        match stdout_capture with
+        | `No_capture -> lazy ""
+        | `Capture fn -> lazy (Stdune.Io.read_file fn)
+      in
+      let actual_stderr =
+        match stderr_capture with
+        | `No_capture
+        | `Merged_with_stdout ->
+          lazy ""
+        | `Capture fn -> lazy (Stdune.Io.read_file fn)
+      in
+      let outputs_are_as_expected =
```

Comment:
It seems safer to replace it with a pattern match on all cases, to make the code robust if we decide to add more variants.

Reply:
(To me, it would be great if we couldn't even compare variants for equality, because that seems like a footgun.)

Reply:
done